zebrad/src/components/mempool/crawler.rs

```diff
@@ -0,0 +1,88 @@
+//! Zebra Mempool crawler.
+//!
+//! The crawler periodically requests transactions from peers in order to populate the mempool.
+
+use std::time::Duration;
+
+use futures::{stream, StreamExt, TryStreamExt};
+use tokio::{sync::Mutex, time::sleep};
+use tower::{BoxError, Service, ServiceExt};
+
+use zebra_network::{Request, Response};
+
+/// The number of peers to request transactions from per crawl event.
+const FANOUT: usize = 4;
+
+/// The delay between crawl events.
+const RATE_LIMIT_DELAY: Duration = Duration::from_secs(75);
+
+/// The mempool transaction crawler.
+pub struct Crawler<S> {
+    peer_set: Mutex<S>,
```

Comment:
Done ([dfb3e5f](https://github.com/jvff/zebra/commit/dfb3e5fad7e54e48332c00f353b4d69304c0d9aa)). I also refactored the error handling, because previously the fan-out would be cancelled as soon as an error occurred ([7fc76c8](https://github.com/jvff/zebra/commit/7fc76c8f4c48feaf73315991ae63a9751ef8d552)).